### PR TITLE
Frontend first step

### PR DESCRIPTION
**Pull request by jbsv**
- React/typescript app creation
- basic move and renaming of files to ease later comparison

### CODE REVIEW

File: web/frontend/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+  "compilerOptions": {
+    "target": "es5",
+    "lib": [
+      "dom",
+      "dom.iterable",
+      "esnext"
+    ],
+    "allowJs": true,
```
Comment (pierluca) [inline web/frontend/tsconfig.json:9]: can't this be set to `false` now ?

**Review by pierluca — COMMENTED** [review]
I've had a quick look at the whole thing.

I'm a bit concerned, in particular, with throwing away all the dependencies (and the package-lock).
In my experience, package versioning in the JS ecosystem is often a mess, and it can take days of trial and error to find the exact combination of dependencies versions that make the project "just work".

As such, I'd rather take an incremental approach to the previous dependencies than start from scratch.

File: web/frontend/package.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
+{
+  "name": "frontend",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
```
[the remark this answers is not in the document]
Comment (jbsv) [reply web/frontend/package.json:5]: yes, at this stage, this files comes straight from the template

File: web/app/tailwind.config.js [hunk 3]
```diff
@@ -1,11 +0,0 @@
...
-  purge: ['./src/**/*.{js,jsx,ts,tsx}', './public/index.html'],
-  darkMode: false, // or 'media' or 'class'
-  theme: {
-    extend: {},
-  },
-  variants: {
-    extend: {},
-  },
-  plugins: [],
-}
```
[the remark this answers is not in the document]
Comment (jbsv) [reply web/app/tailwind.config.js:11]: I didn't know what it's used for :-D
but planned  to look at it later
I ll put it back in

File: web/frontend/package.json [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    "@testing-library/jest-dom": "^5.16.2",
+    "@testing-library/react": "^12.1.2",
+    "@testing-library/user-event": "^13.5.0",
+    "@types/jest": "^27.4.0",
+    "@types/node": "^16.11.22",
+    "@types/react": "^17.0.39",
+    "@types/react-dom": "^17.0.11",
```
[the remark this answers is not in the document]
Comment (jbsv) [reply web/frontend/package.json:12]: patience

File: web/frontend/src/App.test.tsx [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import App from './App';
+
+test('renders learn react link', () => {
```
[the remark this answers is not in the document]
Comment (jbsv) [reply web/frontend/src/App.test.tsx:5]: that's the one created with the template. will be worked on later

File: web/frontend/tsconfig.json (hunk 1 above)
Comment (jbsv) [reply web/frontend/tsconfig.json]: is this compatible with:
 
  "compilerOptions": {
    "paths": {
      "@dedis/kyber": [
        "../cothority/external/**js**/kyber/src"

**Pull request merged**